### PR TITLE
design and action completed (#8)

### PR DESCRIPTION
**Pull request by milg15**
I think this could be move to another file but you can validate me first if the implementation it's good to go 😄

### CODE REVIEW

File: packages/app/src/components/inspector/ColorInputFieldWithPicker.tsx [hunk 1]
```diff
@@ -52,6 +76,72 @@ interface Props {
...
+              rgbaToHsva(sketchColorToRgba(item.value)),
+            );
+
+            return (
+              <Square
+                key={item.do_objectID}
+                color={colorString}
+                selected={selected === item.do_objectID}
+                onClick={() => {
+                  onClick(item.value);
```
Comment (dabbott) [inline packages/app/src/components/inspector/ColorInputFieldWithPicker.tsx:108]: Ok - I don't think I explained this part very well... colors have a property called `swatchID` which we should set when a swatch is selected. For example:

```ts
onClick({
  ...item.value,
  swatchID: item.do_objectID,
});
```

We can then use this to show selection. So for example, `GridProps` could contain a `selectedSwatchId` prop, and then 

```tsx
<Square 
  selected={selectedSwatchId === item.do_objectID} 
  // ... 
/>
```

Then in the `ColorInputFieldWithPicker` we'll render using 

```tsx
<SwatchesSelector
  selectedSwatchId={value.swatchID} 
  // ...
/>
```

Hopefully that's clear - if not let me know

File: packages/app/src/components/inspector/ColorInputFieldWithPicker.tsx [hunk 2]
```diff
@@ -52,6 +76,72 @@ interface Props {
   onChange: (color: Sketch.Color) => void;
 }
 
+interface GridProps {
+  id?: string;
```
Comment (dabbott) [inline packages/app/src/components/inspector/ColorInputFieldWithPicker.tsx:80]: `selectedSwatchId` maybe?

**Review by dabbott — CHANGES_REQUESTED** [review]
Looks good! It's pretty concise. I'm thinking we can leave it all in this file for now.

File: packages/app/src/components/swatches/ColorSwatch.tsx [hunk 3]
```diff
@@ -17,10 +16,7 @@ interface Props {
 }
 
 export default memo(function ColorSwatch({ value }: Props) {
-  const colorString = useMemo(
-    () => hsvaToRgbaString(rgbaToHsva(sketchColorToRgba(value))),
-    [value],
-  );
+  const colorString = useMemo(() => sketchColorToRgbaString(value), [value]);
```
Comment (milg15) [inline packages/app/src/components/swatches/ColorSwatch.tsx:19]: I changed it so it uses the proper function

**Review by dabbott — APPROVED** [review]

**Comment by dabbott** [comment]
Looks great! I'll go ahead and merge it 🚀

**Comment by milg15** [comment]
awesome 🎉

**Pull request merged**